### ISSUE TITLE
python3complete emits FutureWarning

### ISSUE THREAD

**Issue by fancsali** [vim/vim#9981]
### Steps to reproduce

1. Do `import pandas as pd`
2. Try to autocomplete: `pd.<C-X><X-O>`
3. Due to deprecation warnings in the current Pandas version (1.4.1), several `FutureWarning` error messages will be displayed on the bottom of the screen, and one needs to press <ENTER> several times before seeing the actual completions.

### Expected behaviour

The warnings are "swallowed" by the completion script, or at least one won't have to press <ENTER> to accept them.

### Version of Vim

8.2

### Environment

Linux Mint 20.3 Xfce 64-bit

**Comment by brammool** [comment]
Please quote the errors you see.
Also, I assume you edit a Python file?  Precise reproduction steps might be needed.

The python3complete.vim script has no active maintainer.

**Comment by lacygoill** [comment]
Here is a MWE:

Install pandas:

    python3 -m pip install pandas

Then start Vim like this:

    vim -Nu NONE -S <(tee <<'EOF'
        vim9script
        ['import pandas as pd.']->writefile('/tmp/py.py')
        filetype plugin on
        edit /tmp/py.py
        feedkeys("A\<C-X>\<C-O>", 'nxt')
    EOF
    )

This error is given:

    Error detected while processing function python3complete#Complete:
    line   35:
    <string>:134: FutureWarning: pandas.Float64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.

It seems the error can't be caught.  However, it can be silenced:
```diff
diff --git a/runtime/autoload/python3complete.vim b/runtime/autoload/python3complete.vim
index 0635ef824..8b1f5bf36 100644
--- a/runtime/autoload/python3complete.vim
+++ b/runtime/autoload/python3complete.vim
@@ -84,7 +84,7 @@ function! python3complete#Complete(findstart, base)
                 break
             endif
         endwhile
-	execute "py3 vimpy3complete('" . escape(cword, "'") . "', '" . escape(a:base, "'") . "')"
+	execute "silent! py3 vimpy3complete('" . escape(cword, "'") . "', '" . escape(a:base, "'") . "')"
         return g:python3complete_completions
     endif
 endfunction
```

**Comment by chrisbra** [comment]
I have no idea of python3, but this patch seems to prevent it as well:
```
diff --git a/runtime/autoload/python3complete.vim b/runtime/autoload/python3complete.vim
index 0635ef824..8b17ef6b4 100644
--- a/runtime/autoload/python3complete.vim
+++ b/runtime/autoload/python3complete.vim
@@ -91,6 +91,8 @@ endfunction

 function! s:DefPython()
 py3 << PYTHONEOF
+import warnings
+warnings.simplefilter(action='ignore', category=FutureWarning)
 import sys, tokenize, io, types
 from token import NAME, DEDENT, NEWLINE, STRING
```

**Comment by fancsali** [comment]
@chrisbra: Took the liberty to take your code and create a PR with it. Set yourself as the author and myself the humble committer:  #10043.

**Comment by chrisbra** [comment]
thanks that is fine of course :)

**Comment by chrisbra** [comment]
I guess this can be closed now: #10043
